src/dex/jarvis-v6/utils.ts

```diff
@@ -0,0 +1,130 @@
+        callData: poolInterface.encodeFunctionData('feePercentage', []),
+      },
+    ])
+    .flat();
+  const poolDataCalled = await multiContract.methods
+    .aggregate(PoolCallData)
+    .call({}, blockNumber);
+
+  let i = 0;
+  return poolConfigs.map((pool, index) => {
+    return {
+      priceFeed: {
+        usdcPrice: bigIntify(pricesFeedResults[index]),
+      },
+      pool: {
+        feesPercentage: bigIntify(
+          poolInterface.decodeFunctionResult(
+            'feePercentage',
+            poolDataCalled.returnData[i++],
+          )[0],
+        ),
+      },
+    };
+  });
+}
+
+async function _getPricesFeedValue(
+  dexHelper: IDexHelper,
+  poolConfigs: PoolConfig[],
+  priceFeedAddress: Address,
+  blockNumber: number | 'latest',
+) {
+  const pairList = poolConfigs.map(pool =>
+    ethers.utils.formatBytes32String(pool.priceFeedPair),
+  );
+
+  const priceFeedContract = new dexHelper.web3Provider.eth.Contract(
+    SynthereumPriceFeedABI as any,
+    priceFeedAddress,
+  );
+
+  const pricesFeedValues = await priceFeedContract.methods
+    .getLatestPrices(pairList)
+    .call({}, blockNumber);
+
+  return pricesFeedValues;
+}
+
+export function getJarvisPoolFromTokens(
+  srcToken: Token,
+  destToken: Token,
+  poolConfigs: PoolConfig[],
+): PoolConfig | null {
+  const srcAddress = srcToken.address.toLowerCase();
+  const destAddress = destToken.address.toLowerCase();
+  return (
+    poolConfigs.find(pool => {
+      const collateralAddress = pool.collateralToken.address.toLowerCase();
+      const syntheticToken = pool.syntheticToken.address.toLowerCase();
+      return (
+        (srcAddress === collateralAddress && destAddress === syntheticToken) ||
+        (srcAddress === syntheticToken && destAddress === collateralAddress)
+      );
+    }) ?? null
+  );
+}
+
+export function convertToNewDecimals(
+  amount: bigint,
+  currentDecimal: number,
+  desireDecimal: number,
+): bigint {
+  if (currentDecimal === desireDecimal) return amount;
+  const isDecimalIncrease = currentDecimal < desireDecimal;
+  const bigIntPow = getBigIntPow(
+    isDecimalIncrease
+      ? desireDecimal - currentDecimal
+      : currentDecimal - desireDecimal,
+  );
+
+  return isDecimalIncrease ? amount * bigIntPow : amount / bigIntPow;
+}
+
+// This is needed in order to not modify existing logic and use this wrapper
+// to be safe if we receive not cached decimals
+export function getBigIntPow(decimals: number): bigint {
+  const value = BI_POWS[decimals];
+  // It is not accurate to create 10 ** 23 and more decimals from number type
+  return value === undefined ? BigInt(`1${'0'.repeat(decimals)}`) : value;
+}
+
+export function getJarvisSwapFunction(
+  srcToken: Token,
+  pool: PoolConfig,
+): JarvisSwapFunctions {
+  const srcAddress = srcToken.address.toLowerCase();
+  if (srcAddress === pool.collateralToken.address.toLowerCase())
+    return JarvisSwapFunctions.MINT;
+  return JarvisSwapFunctions.REDEEM;
```

Comment:
safer to check if `(srcToken|destToken)` is a `(jFiat|collateralToken)` (see my comment up in call stack)

Reply:
This function is only called when the pool has been already defined. I then supposed that both `srcAddress` and `destAddress` target a valid pool. Only the `MINT` or `REDEEM` info is required in order to know with function to call from the contract.

Reply:
[this](https://github.com/paraswap/paraswap-dex-lib/pull/171#discussion_r964535355) works for me, ignore here